### PR TITLE
Resolving the contribution processor at the time the pending contribution is being retried.

### PR DESCRIPTION
**Pull request by szilardszaloki**
Resolves https://github.com/brave/brave-browser/issues/17766.

Database migration: `v32` `->` `v33`.

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
1. Verify your Rewards wallet with Uphold.
2. Try tipping a Gemini-verified publisher - e.g. [jumde.github.io](https://jumde.github.io).
3. Confirm the tip goes to the pending tips list (in the Summary view on the Rewards panel).
4. Disconnect from Uphold and verify your Rewards wallet with Gemini.
5. Advance your computer clock 3 days.
6. Make sure the tip gets removed from the pending list and succeeds (Gemini currently has a delay on the balance actually being moved from the user's account).
---
Make sure database migration (`v32` `->` `v33`) is successful, that is, there's no `processor` column in the `pending_contribution` table anymore (after upgrading to this version, or with a clean profile).

### CODE REVIEW

File: vendor/bat-native-ledger/test/data/publisher_info_db_v30.sql [hunk 1]
```diff
@@ -205,7 +205,6 @@ CREATE TABLE IF NOT EXISTS "event_log" (
 INSERT INTO "meta" VALUES ('mmap_status','-1'),
  ('version','30'),
  ('last_compatible_version','1');
-INSERT INTO "pending_contribution" VALUES (1,'pub_key', 1.0, 0, 'v01', 0);
```
Comment (zenparsing) [inline vendor/bat-native-ledger/test/data/publisher_info_db_v30.sql:208]: Curious: why was this line removed?

File: vendor/bat-native-ledger/src/bat/ledger/internal/database/database_migration.h [hunk 2]
```diff
@@ -15,6 +15,7 @@ namespace ledger {
 class LedgerImpl;
 
 namespace database {
+extern uint32_t test_target_version;
```
Comment (zenparsing) [inline vendor/bat-native-ledger/src/bat/ledger/internal/database/database_migration.h:18]: Can we put this behind a `ForTesting` method? For example, something like:

```cpp
class DatabaseMigration {
  public:
    static void SetTargetVersionForTesting(uint32_t version);
};
```

File: vendor/bat-native-ledger/test/data/publisher_info_db_v30.sql (hunk 1 above)
Comment (szilardszaloki) [reply vendor/bat-native-ledger/test/data/publisher_info_db_v30.sql:208]: I found a simpler way to check whether the `pending_contribution` table has the `processor` column:
```
EXPECT_TRUE(GetDB()->DoesColumnExist("pending_contribution", "processor"));
```

File: vendor/bat-native-ledger/src/bat/ledger/internal/database/database_migration.h (hunk 2 above)
Comment (szilardszaloki) [reply vendor/bat-native-ledger/src/bat/ledger/internal/database/database_migration.h:18]: Sure, good idea!

**Review by emerick — APPROVED** [review]
LGTM, but agree with Kevin's suggestion for the test method.

**Review by zenparsing — APPROVED** [review]
